src/main/solution_space/path_finder/path_finder_v_3.py

```diff
@@ -134,14 +139,23 @@ def __get_top_n_candidates(top_n: int, nodes_number: int, nodes_numbers_dict: Di
 
         return candidates
 
+    def __choose_best_anon_tree(self, user_anon_tree: AnonTree, anon_trees: List[AnonTree]) -> Optional[AnonTree]:
+        """
+        1. Sort candidates using MeasuredTree
+        2. Return the first candidate
+        """
+        log.info(f'Number of candidates: {len(anon_trees)}\nCandidates ids are {([a_t.id for a_t in anon_trees])}')
+        if len(anon_trees) == 0:
+            return None
+        candidates = list(map(lambda anon_tree: self.get_measured_tree(user_anon_tree, anon_tree), anon_trees))
+        candidates.sort()
+        log.info(f'The best vertex id is {candidates[0].candidate_tree.id}')
+        return candidates[0].candidate_tree
+
     @staticmethod
     def __is_far_from_graph(diffs_from_user_to_goal: int, diffs_from_user_to_graph_vertex: int) -> bool:
         return diffs_from_user_to_graph_vertex / diffs_from_user_to_goal > DISTANCE_TO_GRAPH_THRESHOLD
 
-    @staticmethod
-    def __is_most_of_path_is_done(diffs_from_empty_to_goal: int, diffs_from_user_to_goal: int) -> bool:
```

Comment:
Maybe we should keep this method? We just checked the number of nodes, but maybe we still want to check distance?